File: app/src/main/java/com/relic/domain/models/CommentModel.java
package com.relic.domain.models;

import com.squareup.moshi.JsonClass;

//@JsonClass(generateAdapter = true)
public class CommentModel extends ListingItem {
    public static int UPVOTE = 1;
    public static int DOWNVOTE = -1;
    public static int NOVOTE = 0;
    public static String TYPE = "t1_";

    public String id;
    private String body;
    public String created;
    private int score;

    // post parent id, NOT fullname
    // TODO tbh really need to tidy up consistency with convention
    // TODO I know I used id for fullname of posts, but should be changed to fullname there
    public String parentPostId;
    public String authorFlairText;

    public int platinum;
    public int gold;
    public int silver;

    public boolean isSubmitter;

    public String edited;
    public int depth;
    public int replyCount;
    public String replyLink;

    public String linkTitle;
    public String linkAuthor;

    public CommentModel() {}

    public String getBody() {
    return body == null ? "" : body;
    }

    public void setBody(String body) {
    this.body = body;
    }

    public int getScore() {
    return score;
    }

    public void setScore(int score) {
    this.score = score;
    }

    public float position;

    public boolean isLoadMore() {
      return author == null;
  }

    @Override
    public String getFullName() {
        return "t1_" + id;
    }
}

File: app/src/main/java/com/relic/domain/models/CommentModel.kt
package com.relic.domain.models

import androidx.room.Entity
import com.relic.api.qualifier.ConvertHtml
import com.relic.api.qualifier.Date
import com.relic.api.qualifier.More
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
@Entity
open class CommentModel: ListingItem() {

    @ConvertHtml
    @Json(name = "body_html")
    var bodyHtml: String = ""

    @ConvertHtml
    var body: String = ""

    // id of this comment's direct ancestor (could be a post or comment)
    // this is a fullname (ignore the fact the field says id)
    @Json(name = "parent_id")
    var parentFullname: String = ""

    // fullname of this comment's root post
    // this is a fullname (ignore the fact the field says id)
    @Json(name = "link_id")
    var linkFullname: String? = null

    var authorFlairText: String? = null

    var platinum: Int = 0
    var gold: Int = 0
    var silver: Int = 0

    @Json(name = "is_submitter")
    var isSubmitter: Boolean = false

    var subreddit : String? = null

    @Date
    var edited: java.util.Date? = null

    var depth: Int = 0
    var replyCount: Int = 0
    var replyLink: String? = null

    @Json(name = "link_title")
    var linkTitle: String? = null

    @Json(name = "link_author")
    var linkAuthor: String? = null

    var position: Float = 0F

    val isLoadMore: Boolean
        get() = author == ""

    // use only if this is a "more" comment in which case this will be a list of ids
    @More
    @Json(name = "children")
    var more : List<String>? = null

    companion object {
        var UPVOTE = 1
        var DOWNVOTE = -1
        var NOVOTE = 0
        var TYPE = "t1_"
    }
}
